### PR TITLE
Add Docusaurus help site

### PR DESCRIPTION
**Pull request by Nateowami**
The most interesting file in this change is build.sh, as it grabs all the site content and produce the output.

A number of files were generated by Docusaurus.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1734)

### CODE REVIEW

**Review by Nateowami — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1734)* status: 0 of 12 files reviewed, 7 unresolved discussions

___
*[`help/.gitignore` line 22 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPypH5D3D6m2pzPhBqp:-NPypH5D3D6m2pzPhBqq:b-mpy0fe) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/.gitignore#L22)):*
> ```ignore
> yarn-error.log*
> 
> /s3
> ```

Most of this file was created automatically and I left it. The s3 folder is a temporary folder used by the build script that I added.
___
*[`help/babel.config.js` line 1 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPypPSQBIvc6wEBIcQd:-NPypPSR2N98RWoNWszj:b-vsdbto) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/babel.config.js#L1)):*
> ```JavaScript
> module.exports = {
> ```

This is a Docusaurus-generated file that I did not change.
___
*[`help/sidebars.js` line 1 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPyq-oc84W02oznVzEB:-NPyq-oc84W02oznVzEC:b-tviyn9) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/sidebars.js#L1)):*
> ```JavaScript
> /**
> ```

I did not touch this file.
___
*[`help/docs/getting-started.mdx` line 1 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPyq2pQ2PBb2c2wfEXX:-NPyq2pQ2PBb2c2wfEXY:b-mx5gc4) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/docs/getting-started.mdx#L1)):*
> ```Markdown
> ---
> ```

This is the only "document" that I checked in instead of fetching from Notion. It's the home page, which I wanted to customize more than could be done in just Markdown.
___
*[`help/docs/getting-started.mdx` line 3 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPyqDSH0m8gnEsZFjb8:-NPyqDSH0m8gnEsZFjb9:b1do2o5) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/docs/getting-started.mdx#L3)):*
> ```Markdown
> ---
> slug: /
> sidebar_position: -1
> ```

The files fetched from Notion are given sequential numbers according to their order in Notion, starting at 0. Setting this to -1 ensured it was always first.
___
*[`help/i18n/fr/docusaurus-plugin-content-docs/current/getting-started.mdx` line 8 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPyqPx51xxDHF1pDTC_:-NPyqPx51xxDHF1pDTCa:b4tmgv4) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/i18n/fr/docusaurus-plugin-content-docs/current/getting-started.mdx#L8)):*
> ```Markdown
> import DocCardList from "@theme/DocCardList";
> 
> # Commencer
> ```

This is the only "localized" file I checked in (I translated only this one word), only to demonstrate that things are configured properly and we can drop localized files in the same place and they will be picked up properly. The actual documents haven't been localized, so they all fall back to English.

Once we have the build script fetching from Crowdin we can delete this file.
___
*[`help/src/css/custom.css` line 35 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1734#-NPyqqMDAoq5l2gEp24V:-NPyqqMDAoq5l2gEp24W:btcs2vx) ([raw file](https://github.com/sillsdev/web-xforge/blob/2cd599fdda399d89dc4ce3a7cb9ee02040ff6987/help/src/css/custom.css#L35)):*
> ```CSS
> 
> /* The SF standard gray matches poorly with the dark background when in dark mode, so only apply in light mode */
> html[data-theme='light'] .footer {
> ```

If I recall correctly, this rule was the only change I made to the file.

File: help/build.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+cp ../src/SIL.XForge.Scripture/ClientApp/src/favicon.ico static/img/favicon.ico
+cp ../src/SIL.XForge.Scripture/ClientApp/src/assets/icons/sf.svg static/img/sf.svg
+
+# Download the help files from Notion
+npm install
+npx docu-notion -n $SF_HELP_NOTION_TOKEN -r $SF_HELP_NOTION_ROOT_PAGE_ID
+
+# Needed for the current version of docu-notion; this can be removed once it publishes a new release
+sed -i s/:::📝/:::note/g docs/*.md
+
```
Comment (pmachapman) [inline help/build.sh:16]: Do we want a comment to here to say we are downloading the manual?

**Review by pmachapman — APPROVED** [review]
This looks good.

I was not able to full test the scripts, due to config I do not have access to, but could run parts of it, and see how things correspond to the live site.

Would you be able to document the build process in the Wiki? I notice there are a few environment variables required for `build.sh` to get it to run. The script also contacts "black boxes" like S3 and docu-notion, that took me a bit of digging/tinkering to find out what they contain.

**Comment by Nateowami** [comment]
This has been moved to a different repo: https://github.com/sillsdev/scripture-forge-help

**Pull request closed without merging**